### ISSUE TITLE
Changing provider_instance_size_name do a lower instance succeeds in terraform but does not change the cluster

### ISSUE THREAD

**Issue by ltsonov-cb** [mongodb/terraform-provider-mongodbatlas#683]
Hello. 
I did not find a similar issue and it feels like everything should be working fine.
The TL;DR is that I am trying to downscale a cluster (M50 -> M30) via terraform but it doesn't do anything. 
My guess is that it could be because the cluster's storage was scaled up and is over the M30 limit - but then the provider should not report success in my opinion. 

See below for info. 

### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform version
Terraform v0.13.5

Provider version
mongodb/mongodbatlas v1.2.0 (signed by a HashiCorp partner, key ID 2A32ED1F3AD25ABF)
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_cluster" "cluster" {
  project_id   = local.atlas_project_id
  name         = local.atlas_cluster_name
  cluster_type = "REPLICASET"
  replication_specs {
    num_shards = 1
    regions_config {
      region_name = var.mongodb_atlas_region
      electable_nodes = 3
      priority        = 7
      read_only_nodes = 0
    }
  }

  advanced_configuration {
    javascript_enabled           = false
    minimum_enabled_tls_protocol = "TLS1_2"
  }

  cloud_backup                 = true
  pit_enabled                  = var.enable_continuous_backup
  auto_scaling_disk_gb_enabled = var.enable_auto_scaling_disk
  mongo_db_major_version       = "5.0"

  provider_name               = "AWS"
  provider_instance_size_name = var.provider_instance_size_name

  lifecycle {
    prevent_destroy = true
  }
}
```

### Steps to Reproduce

1. Deploy M50 cluster with configuration above
2. Scale disk on the cluster to 1TB 
3. Change `provider_instance_size_name` to M30
4. Run `terraform plan`
5. Run `terraform apply`
6. Observe cluster

### Expected Behavior

The provider warns that downscaling is not possible or successfully downscales the cluster.

### Actual Behavior

Provider says that all modifications are completed successfully but the cluster remains a M50 size. 

### Debug Output 
Output from `terraform plan` - this is the only change in state:
```
      ~ provider_instance_size_name             = "M50" -> "M30"
```
Output from `terraform apply`:
```
mongodbatlas_cluster.cbcontainers_cluster: Modifying... [<REDACTED>]
mongodbatlas_cluster.cbcontainers_cluster: Modifications complete after 1s [<REDACTED>]
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Additional Context

The cluster in question has 1024 GB (1TB) of storage allocated to it. 
Not sure if this prevents downscaling via Terraform since M30 can't go above 1TB.

The provider should warn about this if possible though - otherwise it looks like the cluster was scaled but it is not.

**Comment by themantissa** [comment]
@ltsonov-cb do you have any logs to provide as requested in our issue template?  Debug is best.  The provider depends on the error messages returned by the API so the logs would, hopefully, show if something was returned and the provide considered it complete anyway.   If that's the case it is something we can address in the provider.  However, the provider is not going to have all of the logic of the system itself so that's why seeing the interaction in the logs is so vital.

**Comment by ltsonov-cb** [comment]
@themantissa attaching gist with the project ID redacted. 

[GIST](https://gist.github.com/ltsonov-cb/5cf211dd51becc1c41796aaac0edea54)

The problem indeed seems to be that the disk on the instance is too large - see the response details below. 
But shouldn't the provider throw an error in this case instead of silently succeeding? 

```
HTTP/2.0 400 Bad Request
Content-Length: 316
Content-Type: application/json
Date: Thu, 24 Feb 2022 09:11:47 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: envoy
Strict-Transport-Security: max-age=31536000; includeSubdomains;
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 109
X-Frame-Options: DENY
X-Mongodb-Service-Version: gitHash=ac979ae4349b7fc9553a734e34c9aeaf751f3680; versionString=v20220216
X-Permitted-Cross-Domain-Policies: none

{
 "detail": "Invalid disk size: Cluster disk size 1024 GB is greater than the maximum disk size of 512 GB for instance size M30.",
 "error": 400,
 "errorCode": "CLUSTER_DISK_SIZE_INVALID",
 "parameters": [
  "Cluster disk size 1024 GB is greater than the maximum disk size of 512 GB for instance size M30"
 ],
 "reason": "Bad Request"
}
```

**Comment by themantissa** [comment]
@ltsonov-cb Thank you for the additional information!  Yes, the provider should catch this error and not respond as it did.  I will file a ticket.

**Comment by themantissa** [comment]
Internal ticket INTMDB-303

**Comment by martinstibbe** [comment]
@ltsonov-cb This fix will be in version 1.3.1 pre release for testing soon

**Comment by themantissa** [comment]
@ltsonov-cb  Pre-release with the fix is out if you'd like to test it. We'll likely release Monday. https://github.com/mongodb/terraform-provider-mongodbatlas/releases/tag/untagged-bcc6453f1acaa220dd7e

**Comment by themantissa** [comment]
1.3.1 is released.

**Comment by ltsonov-cb** [comment]
Thank you for resolving this!
